sklearn/ensemble/_stacking.py

```diff
@@ -104,11 +104,22 @@ def _concatenate_predictions(self, X, predictions):
         """
         X_meta = []
         for est_idx, preds in enumerate(predictions):
-            # case where the estimator returned a 1D array
-            if preds.ndim == 1:
+            # case where the the estimator returned a 1D array
+            if hasattr(preds, "ndim") and preds.ndim == 1:
```

Comment:
Might there be array-likes that we would need to transform but which won't pass this test (e.g. just Python list)?

With the prior array validation, I would say that it is unlikely with work with something else than numpy array or SciPy sparse matrices but I think it's still worth testing.

Reply:
Ok got it but I failed to see how to test that. Do you have any specific suggestions?

Reply:
Looking at `_concatenate_predictions`' occurrences, transforming a python list of int using a class extending `_BaseStacking` should do.

Reply:
In regards to our API, we only have a list or NumPy arrays.

I would be more in favor of having the following:

```python
if isinstance(preds, list):  # multi-label probability predictions
    # add comment regarding the expected shape
    for pred in preds:
        # arbritrary drop the first class probabilities to avoid collinearity
        X_meta.append(pred[:, :1])
elif pred.ndim == 1:
    ...
elif (
    self.stack_method_[est_idx] == "predict_proba"
    and len(classes_) == 2
):
    ...
else:
    ...
```